### ISSUE TITLE
breaking when using gulp-autoprefixer

### ISSUE THREAD

**Issue by sQVe** [dlmanning/gulp-sass#267]
Heya,

I've been using a gulp task to build my sass files, code snippet available below, and since the latest update of gulp-sass it no longer works. It halts and the sass task is never finished - no error thrown. If I remove **gulp-autoprefixer** it works though.

A bit unsure if this is an issue with gulp-sass or something I should take with **gulp-autoprefixer**, but **gulp-sass** is the lib that broke it with the update so.

```coffeescript
# Dependencies.
gulp         = require "gulp"
gulpif       = require "gulp-if"
sass         = require "gulp-sass"
sourcemaps   = require "gulp-sourcemaps"
autoprefixer = require "gulp-autoprefixer"
minifyCss    = require "gulp-minify-css"
browserSync  = require "browser-sync"
onError      = require "../helpers/onError"


gulp.task "sass", ->
  gulp.src "sass/**/*.{sass,scss}"
    .pipe sourcemaps.init()
    .pipe sass()
    .on "error", onError
    .pipe autoprefixer()
    .pipe sourcemaps.write()
    .pipe gulp.dest "./build"
    .pipe browserSync.reload stream: true
```

**Comment by Keats** [comment]
The .on seems to be on the pipe(sass()) instead of on sass(). 
I use autoprefixer in all of my projects and they all work, so most likely something is wrong in the gulpfile

**Comment by sQVe** [comment]
@Keats That doesn't change anything, sadly. If I simplify it even further like this:

```coffeescript
gulp.task "sass", ->
  gulp.src "sass/**/*.{sass,scss}"
    .pipe sass().on "error", ->
      console.log "error"
    .pipe autoprefixer()
    .pipe gulp.dest "./build"
```
It still never finishes. I'm using **gulp-sass** version *2.0.0*, **1.3.3** works perfectly.

**Comment by Keats** [comment]
Can you try this project of mine https://github.com/Keats/snowflake which is basically the same task as yours almost and tell me if it's working or not?

**Comment by sQVe** [comment]
@Keats Seems to be an issue here with Gulp V4 tho. Keep getting this:

```
✔ ~/code/snowflake [master|✔] 
13:27 $ gulp sass
[13:27:24] Using gulpfile ~/code/snowflake/gulpfile.js
/usr/lib/node_modules/gulp/bin/gulp.js:129
    gulpInst.start.apply(gulpInst, toRun);
                  ^
TypeError: Cannot read property 'apply' of undefined
    at /usr/lib/node_modules/gulp/bin/gulp.js:129:19
    at doNTCallback0 (node.js:415:9)
    at process._tickCallback (node.js:344:13)
    at Function.Module.runMain (module.js:474:11)
    at startup (node.js:124:18)
    at node.js:959:3
```

**Comment by Keats** [comment]
Try `npm run gulp watch`
I don't install gulp globally

**Comment by sQVe** [comment]
@Keats That works for me...

I tried compiling your `/scss` folder with my **sass** gulp task. It seems like there is something in my sass files that makes it break on **v2.0.0**. I'm using Bourbon and Neat, might be something there I guess.

**Comment by sQVe** [comment]
@Keats Looked into this further and started commenting out some of my `imports`... If I remove:
```
@import url("http://fonts.googleapis.com/css?family=Roboto:100,100italic,400,400italic,700,700italic,900italic,900,300,300italic");
```
it compiles fine with autoprefixer.

**Comment by Keats** [comment]
Hmm that means it's not a gulp-sass issue then, more of a libsass one.
If you remove autoprefixer and keep the import does it work ?

**Comment by sQVe** [comment]
@Keats Yup.

**Comment by Snugug** [comment]
This is probably related to https://github.com/sass/libsass/issues/1145.

That being said, I am unable to recreate your issue using the latest Gulp Sass and the following reduced use case:

**Sass**
```scss
@import url("http://fonts.googleapis.com/css?family=Roboto:100,100italic,400,400italic,700,700italic,900italic,900,300,300italic");

.foo {
  display: flex;
  color: red;
}
```

**Gulpfile**
```javascript
var gulp = require('gulp'),
    autoprefixer = require('gulp-autoprefixer'),
    sass = require('gulp-sass');

gulp.task('sass', function () {
  gulp.src('./sass/**/*.scss')
    .pipe(sass().on('error', sass.logError))
    .pipe(autoprefixer({
      browsers: ['last 2 versions'],
      cascade: false
    }))
    .pipe(gulp.dest('./css'));
});
```

Please work on pairing down all of your Sass until you find the bare minimum that's causing this on a clean run.

**Comment by sibsfinx** [comment]
Got the same issue and ended up with autoprefixer shell task
```
npm install --save-dev autoprefixer
```

```
shell = require('gulp-shell')

options.autoprefixer =
  browsers: ['Explorer 10', 'Chrome 31', 'Firefox 35', 'Opera 27', 'Safari 7', 'iOS 7', 'Android 4.1', 'ChromeAndroid 41', 'OperaMini 8']
  cascade: false

gulp.task "sass:prefix", shell.task([
  'npm run autoprefixer ' + './dist/stylesheets/*.css ' + '-b '+ '"' + options.autoprefixer.browsers.join(', ').toLowerCase() + '"'
])
```

**Comment by piotrd** [comment]
The error still occurs, although it's claimed to be fixed in libsass 3.2.1 (https://github.com/sass/libsass/issues/1145#issuecomment-97661247). The current `gulp-sass` claims to use `node-sass` 3.0.0, which claims to use libsass 3.2.2 :-/

**Comment by Keats** [comment]
@piotrd does that still occur?

**Comment by xzyfer** [comment]
I believe is fixed in v2.3.1. If the issue persists with latest gulp-sass please open an update issue.
